src/app/modules/shared/services/breadcrumb.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import {
+  Route,
+  ActivatedRouteSnapshot,
+} from '@angular/router';
+import { Observable, of } from 'rxjs';
+
+export interface Breadcrumb {
+  displayName: string;
+  url: string;
+  route?: Route;
+}
+
+@Injectable()
+export class BreadcrumbService {
+  constructor() { }
```

Comment:
no more event emitter and no more declarative logic here. This service is pure, without needing any external dependencies via dependency injection